object/object.go

```diff
@@ -117,18 +185,63 @@ func ArrayEquals(left, right []Object) bool {
 	return true
 }
 
-func MapEquals(left, right Map) bool {
-	if len(left) != len(right) {
+func MapEquals(left, right *Map) bool {
+	if len(left.kv) != len(right.kv) {
 		return false
 	}
-	for k, v := range left {
-		if !Equals(v, right[k]) {
+	for i, kv := range left.kv {
+		if !Equals(kv.Key, right.kv[i].Key) {
+			return false
+		}
+		if !Equals(kv.Value, right.kv[i].Value) {
 			return false
 		}
 	}
 	return true
 }
 
+func CompareKeys(a, b KV) int {
+	return Cmp(a.Key, b.Key)
+}
+
+func (m *Map) Get(key Object) (Object, bool) {
+	kv := KV{Key: key}
+	i, ok := slices.BinarySearchFunc(m.kv, kv, CompareKeys)
```

Comment:
Maybe a comment explaining why a binarysearch is important, it's not a well-known method

Reply:
considering not exporting it also as it's barely used outside